docs/modules/operation/pages/deep-dive/geographical-map/introduction.adoc

```diff
@@ -41,3 +41,57 @@ A cluster's marker displays as the color of the highest severity of the nodes th
 | Critical
 | image::geographical-map/severity_critical.png[]
 |===
+
+== Tile providers
+
+Tile providers are servers that serve up geographical map tiles which are used by the geographical map.
+By default, OpenNMS will use the publicly-available *OpenStreetMap* and *OpenTopoMap*.
+
+However, in cases where you wish to use your own map server, for example a local server inside a firewall, you can specify a user-defined tile provider.
+
+To select between tile providers, click this icon image:geographical-map/tile-provider-selector.png[width=64] on the geographical map.
+
+Then select a provider from the list:
+
+image:geographical-map/tile-provider-list.png[]
+
+In `etc/opennms.properties`, use the following configuration. In this case the local tile provider is found at `192.168.0.99`.
+
+Set `gwt.openlayers.name` to provide a display name, and specify `gwt.openlayers.userDefinedAsDefault=true` if you want this tile provider to be the default (displays as the first one, and loads by default instead of the built-in providers).
+
+```
+###### MAPPING AND GEOCODING ######
+
+# The tile server URL to use for OpenLayers.  This can be any mapnik-style tile server URL.
+# (Sorry, no support for multiple URLs yet.)
+# If 'gwt.openlayers.userDefinedAsDefault' is true, use this tile server as the default one on the Geographical Map page
+# OpenNMS default tile server
+
+gwt.openlayers.name=My Street Map
+gwt.openlayers.url=http://192.168.0.99/{z}/{x}/{y}.png
+gwt.openlayers.options.attribution=Map data &copy; <a tabindex="-1" target="_blank" href="http://192.168.0.99/copyright">My Street Map</a> contributors under <a tabindex="-1" target="_blank" href="192.168.0.99/licenses/odbl/">ODbL</a>, <a tabindex="-1" target="_blank" href="http://192.168.0.99/licenses/by-sa/2.0/">CC BY-SA 2.0</a>
+gwt.openlayers.userDefinedAsDefault=true
+```
+
+In addition, if you supply a user-defined tile provider, you need to update the `'Content-Security-Policy'` of your OpenNMS instance to ensure it is not blocked by the browser.
+
+In `jetty-webapps/opennms/WEB-INF/web.xml`, search for this section.
+Update the `param-value` under *Sets the header value* to include your tile provider's URL in the `img-src` section. In the example below, the URL `http://192.168.0.99` has been added.
```

Comment:
```suggestion
. Update the `param-value` under *Sets the header value* to include your tile provider's URL in the `img-src` section. 
The example below adds the URL `http://192.168.0.99`.
```

Reply:
I updated the hierarchical structure as follows, let me know if it's ok:

--------

== Tile providers

Tile providers are services...

However, ...

To select among tile providers, follow these steps:

1. Click the tile...

2. Select a provider... 

[image of radio button menu]

=== Tile provider configuration

To configure a user-defined...

1. In `etc/opennms.properties` ...

2. Set `gwt.openlayers.name` ...

[MAPPING AND GEOCODING section from `opennms.properties`]

3. In addition, if you supply...

  a. In `jetty-webapps/opennms/WEB-INF/web.xml`, ...

  b. Update the `param-value` ...

[code snippet from `web.xml`]